### ISSUE TITLE
Custom command handler

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/quidem#18]
Allow providing a custom command handler to quidem at run time. By invoking quidem with a particular interface, I could run scripts that have `!foo` and `!bar` commands in them.

This will allow people to extend quidem's assertion-driven testing approach to new problems without having to wait for a new quidem release. If their command is generally applicable, it could later be added to the product.

It would be useful for testing [Calcite's Babel SQL parser](https://issues.apache.org/jira/browse/CALCITE-2280). I could write a script like this:

```
# CONNECT BY only works on Oracle
SELECT lpad(' ',level*2,' ') || ename ename, empno, mgr 
FROM emp 
START WITH mgr IS NULL 
CONNECT BY PRIOR empno = mgr;
select(list: as(concat(lpad(' ', *(level(), integer(2)), ' '), colRef(EMP, ENAME)), ENAME),
  from: tabRef(SCOTT, EMP),
  connectBy: eq(prior(colRef(EMP, EMPNO)), colRef(EMP, MGR))
  startWith: isNull(colRef(EMP, MGR)))
!explain-validated-on oracle
```

I'd need to write my own `explain-validated-on` command that checks/replaces its expected output in the same way that the built-in `ok` command does.

**Comment by julianhyde** [comment]
Under development (almost complete) in 8b0f9378a05c8fec95c454deacaed8819e2cda88.

**Comment by julianhyde** [comment]
Fixed in b848091.
